test/unit/org/apache/cassandra/db/commitlog/CommitlogShutdownTest.java

```diff
@@ -85,15 +87,23 @@ public void testShutdownWithPendingTasks() throws Exception
         // force creating several commitlog files
         for (int i = 0; i < 10; i++)
         {
-            CommitLog.instance.add(m);
+            m.apply();
         }
 
-        // schedule discarding completed segments and immediately issue a shutdown
-        TableId tableId = m.getTableIds().iterator().next();
-        CommitLog.instance.discardCompletedSegments(tableId, CommitLogPosition.NONE, CommitLog.instance.getCurrentPosition());
-        CommitLog.instance.shutdownBlocking();
+        StorageService.instance.drain();
 
-        // the shutdown should block until all logs except the currently active one and perhaps a new, empty one are gone
-        Assert.assertTrue(new File(DatabaseDescriptor.getCommitLogLocation()).listFiles().length <= 2);
+        List<CommitLogSegment> segmentsToCheck = new ArrayList<>(CommitLog.instance.segmentManager.getActiveSegments());
+        // remove the last, potentially active segment from the check
+        if (!segmentsToCheck.isEmpty())
+            segmentsToCheck.remove(segmentsToCheck.size() - 1);
```

Comment:
So to test my understanding, we're saying that at worst we'll allow accumulation of 1 inactive CL Segment on node shutdown here. If a node were cycling up and down and in the pathological worst-case of having a single segment left alive, this would no longer accumulate (the problem in CASSANDRA-13123) so we would then instead only have the 1 CLSegment to contend with, replay on restart, and keep on moving.